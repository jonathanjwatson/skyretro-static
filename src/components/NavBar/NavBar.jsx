
import NavBarLogo from "../../assets/images/sky_retro_logo.png";

const NavBar = (props) => {

  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <span className="navbar-brand" >
        <img
          src={NavBarLogo}
          alt="Sky Retro Logo"
          style={{ maxHeight: "1.5rem" }}
        />
        <span> </span>Sky Retro
      </span>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarText"
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
    </nav>
  );
};

export default NavBar;
