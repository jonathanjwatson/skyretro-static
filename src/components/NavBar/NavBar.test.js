import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import NavBar from "./NavBar";

afterEach(cleanup);

const props = {
  signOut: () => {},
};

describe("Renders the NavBar correctly", () => {
  const history = createMemoryHistory();
  it("renders", () => {
    render(
      <Router history={history}>
        <NavBar {...props} />
      </Router>
    );
  });
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Router history={history}>
        <NavBar {...props} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
