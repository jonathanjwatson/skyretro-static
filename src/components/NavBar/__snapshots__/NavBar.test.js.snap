// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the NavBar correctly matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <img
        alt="Sky Retro Logo"
        src="sky_retro_logo.png"
        style="max-height: 1.5rem;"
      />
      <span>
         
      </span>
      Sky Retro
    </a>
    <button
      aria-controls="navbarText"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarText"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarText"
    >
      <ul
        class="navbar-nav mr-auto"
      />
      <a
        class="btn btn-primary"
        href="/login"
        id="login"
      >
        Sign In
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
